File: clima/frontend/src/types/clima.ts
/**
 * Tipos do domínio — espelham o contrato da nossa API de clima.
 */

export interface Localizacao {
  nome: string;
  pais: string | null;
  latitude: number;
  longitude: number;
}

export interface ClimaAtual {
  temperatura: number;
  umidade: number;
  vento: number;
  descricao: string;
  icone: string;
}

export interface PrevisaoDia {
  data: string;
  temp_min: number;
  temp_max: number;
  descricao: string;
  icone: string;
}

export interface Clima {
  local: Localizacao;
  atual: ClimaAtual;
  previsao: PrevisaoDia[];
}

File: clima/backend/app/schemas/clima.py
"""Schemas Pydantic — formato da resposta da nossa API ao front-end."""

from typing import List

from pydantic import BaseModel


class Localizacao(BaseModel):
    """Dados da cidade localizada."""

    nome: str
    pais: str | None = None
    latitude: float
    longitude: float


class ClimaAtual(BaseModel):
    """Condições atuais do tempo."""

    temperatura: float
    umidade: int
    vento: float
    descricao: str
    icone: str


class PrevisaoDia(BaseModel):
    """Previsão resumida de um dia."""

    data: str
    temp_min: float
    temp_max: float
    descricao: str
    icone: str


class ClimaResponse(BaseModel):
    """Resposta completa: localização + clima atual + previsão."""

    local: Localizacao
    atual: ClimaAtual
    previsao: List[PrevisaoDia]
